docs/design_doc.md

```diff
@@ -0,0 +1,443 @@
+
+**`SpatialData`**
+
+The `SpatialData` object contains a set of Elements to be used for analysis. Elements contained within a `SpatialData` object must be able to share a single Region Table. Future work may extend `SpatialData` to allow multiple tables (see discussion [here](https://github.com/scverse/spatialdata/issues/43)). All Elements within a `SpatialData` object can be queried, selected, and saved via the `SpatialData` objects.
+
+**`NGFFStore`**
+
+_Note: previously we used the term `SpatialDataContainer`; the new naming is `NGFFStore`._
+
+The `NGFFStore` is an object representing the on-disk layout of a dataset. The `NGFFStore` parses the files to determine what data are available to be loaded. Initial implementations will target a single Zarr file on disk, but future implementations may support reading from a collection of files.
+
+### Elements
+
+We model a spatial dataset as a composition of distinct element types. The elements correspond to:
+
+-   Pixel-based images
+-   Regions of interest
+    -   Shapes (such as polygons, circles, ...)
+    -   Pixel masks (such as segmentation masks), aka Labels
+-   Points (such as transcript locations, point clouds, ...)
+-   Tables of annotations (initally, these are annotations on the regions of interest)
+
+Each of these elements should be useful by itself, and in combination with other relevant elements. All elements are stored in the Zarr container in hierarchy store that MAY be flat. `SpatialData` will nonetheless be able to read arbitrary hierarchies and make sense of them based on Zarr groups metadata and coordinate systems.
+
+By decomposing the data model into building blocks (i.e. Elements) we support operations that do not otherwise fit this model. For instance, we may have data where regions have not been defined yet, or are just dealing with images and points.
+
+#### Assumptions
+
+_SpatialData_ closely follows the OME-NGFF specifications and therefore much of its assumptions are inherited from it. Extra assumptions will be discussed with the OME-NGFF community and adapted to the community-agreed design. The key assumptions are the following:
+
+-   `Images`, `Labels`, `Points`, and `Shapes` MUST have one or more _coordinate systems_ and _coordinate transforms_.
+-   `Tables` CAN NOT have a _coordinate system_ or _coordinate transforms_.
+-   `Labels` and `Shapes` are both instances of `Regions`.
+-   `Regions` are `Elements` and they MAY be annotated by `Tables`.
```

Comment:
think important to clarify here re `Points`. If we unify them as `Circles` and `Points` than we should add that `Points` are instanced of `Regions`.
Also below, since we have only one `Table` then `Points` CANNOT be annotated by `Tables.`

Reply:
ok I now read that shapes will be also circles (I like that) hence the first part of the comment can be discarded. The part about points not being annotated by tables though remains.

Reply:
Good catch. Updated to say that Points MAY NOT be annotated by tables.